### PR TITLE
Use Treq for Twisted integration

### PR DESCRIPTION
**Pull request by zgoda-mobica**
[Treq](https://github.com/twisted/treq) is requests-like library for Twisted made by Twisted developers. This branch contains complete rewrite of Twisted support code that uses Treq for web requests.

### CODE REVIEW

**Comment by brianr** [comment]
Thanks @zgoda-mobica ! We will review this in the coming week.

File: rollbar/__init__.py [hunk 1]
```diff
@@ -1246,31 +1193,25 @@ def _send_payload_twisted(payload, access_token):
...
 
+    def post_data_cb(data, resp):
+        resp._content = data
+        _parse_response(path, settings['access_token'], payload, resp)
+
+    def post_cb(resp):
+        r = requests.Response()
+        r.status_code = resp.code
+        r.headers.update(resp.headers.getAllRawHeaders())
+        return treq.content(response).addCallback(post_data_cb, r)
```
Comment (ezarowny) [inline rollbar/__init__.py]: This is breaking for me here. Did you mean to write:

```python
return treq.content(r).addCallback(post_data_cb, r)
```

or

```python
return treq.content(Response).addCallback(post_data_cb, r)
```

File: rollbar/__init__.py [hunk 2]
```diff
@@ -1246,31 +1193,25 @@ def _send_payload_twisted(payload, access_token):
...
 
 
-@inlineCallbacks
+
 def _post_api_twisted(path, payload, access_token=None):
-    headers = {'Content-Type': ['application/json']}
 
+    def post_data_cb(data, resp):
+        resp._content = data
+        _parse_response(path, settings['access_token'], payload, resp)
```
Comment (ezarowny) [inline rollbar/__init__.py]: `settings` should be `SETTINGS`

**Comment by ezarowny** [comment]
Merged in #109

**Pull request closed without merging**